addons/point_of_sale/models/pos_order.py

```diff
@@ -933,6 +948,35 @@ def _launch_stock_rule_from_pos_order_lines(self):
                 pickings_to_confirm.action_confirm()
         return True
 
+    @api.depends('product_id', 'qty')
+    def _compute_cost(self):
+        for line in self:
+            if not line.product_id:
+                line.cost = 0.0
+            else:
+                product = line.product_id
+                product_cost = product.standard_price
+                if not product_cost:
+                    line.cost = 0.0
+                else:
+                    line.cost = abs(line.qty) * product.cost_currency_id._convert(
+                        from_amount=product_cost,
+                        to_currency=line.currency_id,
+                        company=line.company_id or self.env.company,
+                        date=line.order_id.date_order or fields.Date.today(),
+                        round=False,
+                    )
+
+    @api.depends('price_subtotal', 'cost')
```

Comment:
Because of the stored field in `margin` (reason: https://github.com/odoo/odoo/pull/66689/files#r630837163), it is better to separate them to not always recompute the margin whenever we access the view